text/0470-aurora-serverless-v2.md

```diff
@@ -0,0 +1,304 @@
+# Aurora Serverless v2 support
+
+* **Original Author(s):**: @pahud
+* **Tracking Issue**: #470
+* **API Bar Raiser**: @{BAR_RAISER_USER}
+
+Allow users to create Amazon Aurora Serverless v2 instances with the `aws-rds` module.
+
+## Working Backwards
+
+> This section should contain one or more "artifacts from the future", as if the
+> feature was already released and we are publishing its CHANGELOG, README,
+> CONTRIBUTING.md and optionally a PRESS RELEASE. This is the most important
+> section of your RFC. It's a powerful thought exercise which will challenge you
+> to truly think about this feature from a user's point of view.
+>
+> Choose *one or more* of the options below:
+>
+> * **CHANGELOG**: Write the changelog entry for this feature in conventional
+>   form (e.g. `feat(eks): cluster tags`). If this change includes a breaking
+>   change, include a `BREAKING CHANGE` clause with information on how to
+>   migrate. If migration is complicated, refer to a fictional GitHub issue and
+>   add its contents here.
+
+feat(rds): Aurora Serverless v2 support
+
+>
+> * **README**: If this is a new feature, write the README section which
+>   describes this new feature. It should describe the feature and walk users
+>   through usage examples and description of the various options and behavior.
+>
+
+## Aurora Serverless v2 support
+
+Aurora Serverless v2 is an on-demand, autoscaling configuration for Amazon Aurora.
+Aurora Serverless v2 helps to automate the processes of monitoring the workload and
+adjusting the capacity for your databases. Capacity is adjusted automatically based on
+application demand. Read[Using Aurora Serverless v2](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless-v2.html) for more details.
+
+### Using Aurora Serverless v2 for existing provisioned workloads
+
+Aurora Serverless v2 allows you to add one or more Aurora Serverless v2 DB instances to
+the existing cluster as reader DB instances. Make sure the cluster engine is compatible with
+Aurora Serverless v2 and add the `serverlessV2Scaling` before adding serverless instances into
+the cluster. Please note some AWS Regions may not support this feature. Check out the
+supportedengines and Region availability in this [document](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.Aurora_Fea_Regions_DB-eng.Feature.ServerlessV2.html).
+
+```ts
+const cluster = new rds.DatabaseCluster(this, 'Database', {
+  ...
+  instances: 2,
+  // make sure the engine is compatible with serverless v2.
+  engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_02_1 }),
+  // required before you are allowed to add serverless instances
+  serverlessV2Scaling: {
+    maxCapacity: 4,
+    minCapacity: 0.5,
+  },
+});
+// add a serverless reader
+cluster.addServerlessInstance('InstanceId')
```

Comment:
The `serverlessV2Scaling` and `instances` is confusing to me. I believe as a customer this is too much to know and does not reflect intent in the API. 

For instance, what does `instances` mean if I have `serverlessV2Scaling` mentioned. I would assume those are two serverless v2 instances being created by the construct. But these are provisioned instances right?

And we are also assuming the customer to know that there is no way to provision `serverlessInstance: 1` in the API and we are doing it via `addServerlessInstance()`. Let me know if I am getting this right?

And as you mentioned restrictions on region and database engines that are supported by v2. There should either be validations or defaults around it if we are opting for v2 and this responsibility should not lie with the customer to configure.